ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
+            if (!remoteRawData.Exists)
+            {
+                throw new Exception($"SECDataConverter.Process(): Raw data {remoteRawData} not found. No process can be done.");
+            }
+
+            Log.Trace($"SECDataConverter.Process(): Copying raw data locally...");
+
+            var localRawData = remoteRawData.CopyTo(Path.Combine(Path.GetTempPath(), remoteRawData.Name));
+            var extractDataPath = Path.Combine(Path.GetTempPath(), formattedDate);
+
+            Log.Trace($"SECDataConverter.Process(): Extract raw data...");
+
+            using (var data = localRawData.OpenRead())
+            {
+                using (var archive = TarArchive.CreateInputTarArchive(new GZipInputStream(data)))
+                {
+                    Directory.CreateDirectory(extractDataPath);
+                    archive.ExtractContents(extractDataPath);
+
+                    Log.Trace($"SECDataConverter.Process(): Extracted SEC data to path {extractDataPath}");
+                }
+            }
+
+            var tickerList = _tickerFolder.EnumerateDirectories().AsParallel()
+                .Where(d => d.EnumerateFiles($"{formattedDate}*").Any())
+                .Select(d => d.Name);
+
+            Log.Trace($"SECDataConverter.Process(): Start processing..."); 
+            // For the meantime, let's only process .nc files, and deal with correction files later.
             Parallel.ForEach(
-                Directory.GetFiles(RawSource, "*.nc.tar.gz", SearchOption.AllDirectories).ToList(),
-                rawFile =>
+                Directory.GetFiles(extractDataPath, "*.nc", SearchOption.AllDirectories),
+                rawReportFilePath =>
                 {
-                    // GetFileNameWithoutExtension only strips the first extension from the name.
-                    var fileDate = Path.GetFileName(rawFile).Split('.')[0];
-                    var extractDataPath = Path.Combine(RawSource, fileDate);
+                    var factory = new SECReportFactory();
+                    var xmlText = new StringBuilder();
 
-                    DateTime currentDate;
-                    if (!DateTime.TryParseExact(fileDate, "yyyyMMdd", CultureInfo.InvariantCulture, DateTimeStyles.None, out currentDate))
-                    {
-                        throw new Exception($"Unable to parse date from file {rawFile}. Filename we attempted to parse: {fileDate}");
-                    }
-                    
-                    // Only process files within start and end bounds
-                    if (currentDate < startDate || currentDate > endDate)
-                    {
-                        return;
-                    }
+                    // We need to escape any nested XML to ensure our deserialization happens smoothly
+                    var parsingText = false;
 
-                    using (var data = File.OpenRead(rawFile))
+                    foreach (var line in File.ReadLines(rawReportFilePath))
```

Comment:
Nice usage of `File.ReadLines` 👍 to stream the file line-by-line instead of sucking the whole thing up into RAM.